feat: gitlab fetch (#229)
bomctl/bomctl: README.md

bomctl fetch https://github.com/bomctl/bomctl
```

> Or you can use the following syntax to fetch an SBOM from a GitLab repository through the [DependencyListExport web API](https://docs.gitlab.com/ee/api/dependency_list_export.html). Authorization for this command is configured by assigning the value of your GitLab token to the `BOMCTL_GITLAB_TOKEN` environment variable.

Reviewer: Avoid "you" statements in documentation

```suggestion
An SBOM may also be fetched from a GitLab repository through the [DependencyListExport web API](https://docs.gitlab.com/ee/api/dependency_list_export.html) by using the following URL format. Authorization for this command is configured by assigning the value of your GitLab token to the `BOMCTL_GITLAB_TOKEN` environment variable.
```

```shell

> bomctl fetch https://www.gitlab.com/PROJECT/REPOSITORY@BRANCH

Reviewer 2: Does this work? based on the parse function and my other comment, shouldn't this be:

`bomctl fetch https://www.gitlab.com/PROJECT/REPOSITORY#BRANCH`

Author: We're using the fragment when we parse the provided URL string into a `netutil.URL` with the `Parse` method. The user is still expected to use the `@` delimiter, but internally, we're using the fragment because there's no other field in `netutil.URL`.

Reviewer: `GitRef` seems perfectly adequate to me
